### PR TITLE
Add New OSS::Tag component

### PR DESCRIPTION
**Pull request by sBenjamin270495**
### What does this PR do?

Add OSS::Tag component with his documentation and test

Related to: https://github.com/upfluence/backlog/issues/1684

### What are the observable changes?
TagScreen

### Good PR checklist

- [x] Title makes sense
- [x] Is against the correct branch
- [x] Only addresses one issue
- [x] Properly assigned
- [x] Added/updated tests
- [x] Added/updated documentation
- [ ] Migrated touched components to Glimmer Components
- [x] Properly labeled

### CODE REVIEW

File: addon/components/o-s-s/tag.ts [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  }
+
+  get hasEllipsis(): boolean {
+    if (!this.args.hasEllipsis) {
+      return false;
+    }
+    return this.args.hasEllipsis;
+  }
+
+  get computedClass() {
```
Comment (Miexil) [inline addon/components/o-s-s/tag.ts:49]: Missing getter return value

File: addon/components/o-s-s/tag.stories.js [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+  title: 'Components/OSS::Tag',
+  component: 'tag',
+  argTypes: {
+    skin: {
+      description: 'Adjust appearance',
+      table: {
+        type: {
+          summary: SkinTypes.join('|')
+        },
+        defaultValue: { summary: 'regular' }
```
Comment (JulienVannier66) [inline addon/components/o-s-s/tag.stories.js:15]:
```suggestion
        defaultValue: { summary: 'primary' }
```
`regular` does not exist in `SkinTypes`

File: addon/components/o-s-s/tag.stories.js [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+const defaultArgs = {
+  skin: 'default',
+  label: 'Label',
+  icon: 'far fa-thumbs-up',
+  hasEllipsis: 'false'
+};
+
+const Template = (args) => ({
+  template: hbs`
+    <div style="padding: 2em; {{if (eq this.theme 'dark') 'background-color: #060666'}}">
```
Comment (JulienVannier66) [inline addon/components/o-s-s/tag.stories.js:62]: Do you really need this `div`?

**Review by Miexil — APPROVED** [review]
lgtm

File: addon/components/o-s-s/tag.ts [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+      return SkinDefinition.primary;
+    }
+    return SkinDefinition[this.args.skin as SkinType] ?? SkinDefinition.primary;
+  }
+
+  get hasEllipsis(): boolean {
+    if (!this.args.hasEllipsis) {
+      return false;
+    }
+    return this.args.hasEllipsis;
```
Comment (phndiaye) [inline addon/components/o-s-s/tag.ts:46]: `return this.args.hasEllipsis || false`

**Review by phndiaye — APPROVED** [review]

File: addon/components/o-s-s/tag.hbs [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+  {{#if @label}}
+    <div class='text-style-semibold fx-row'>
+      {{#if @hasEllipsis}}
+        <span>{{@label}}</span>
+      {{else}}
+        {{@label}}
+      {{/if}}
+    </div>
+  {{/if}}
+</div>
```
Comment (Miexil) [inline addon/components/o-s-s/tag.hbs:15]: Missing newline at the end of the file

File: addon/components/o-s-s/tag.ts [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+export default class OSSTag extends Component<OSSTagArgs> {
+  constructor(owner: unknown, args: OSSTagArgs) {
+    super(owner, args);
+
+    if (!args.icon && !args.label) {
+      throw new Error('[component][OSS::Tag] You must pass either a @label or an @icon argument.');
+    }
+  }
+
+  get tooltip() {
```
Comment (Miexil) [inline addon/components/o-s-s/tag.ts:34]: Isn't this an unused getter ?

File: addon/components/o-s-s/tag.ts [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+    return this.args.hasEllipsis ? this.args.label : null;
+  }
+
+  get skin(): string {
+    if (!this.args.skin) {
+      return SkinDefinition.primary;
+    }
```
Comment (Miexil) [inline addon/components/o-s-s/tag.ts:41]: You can remove this as you have a default value return underneath if SkinDefinition[this.args.skin] does not match anything

**Review by phndiaye — APPROVED** [review]

File: addon/components/o-s-s/tag.hbs [hunk 8]
```diff
@@ -0,0 +1,15 @@
+<div class={{this.computedClass}} ...attributes>
+  {{#if @icon}}
+    <i class={{@icon}}></i>
+  {{/if}}
+
+  {{#if @label}}
+    <div class='text-style-semibold fx-row'>
```
Comment (Miexil) [inline addon/components/o-s-s/tag.hbs:7]: Use double quotes in template files 😇

**Review by Miexil — APPROVED** [review]

**Review by JulienVannier66 — APPROVED** [review]

**Pull request merged**